packages/libs/src/services/web3Manager/Web3Manager.ts

```diff
@@ -125,8 +124,19 @@ export class Web3Manager {
     this.web3 = web3
   }
 
+  usingInternalWeb3(): boolean {
+    return (this.web3Config && !this.web3Config.useExternalWeb3 && this.web3Config.internalWeb3Config?.web3ProviderEndpoints.length > 0)
+  }
+
   setDiscoveryProvider(discoveryProvider: DiscoveryProvider) {
     this.discoveryProvider = discoveryProvider
+    // if internal web3, use discprov
+    if (this.usingInternalWeb3()) {
+      const gatewayweb3endpoint = this.web3Config.internalWeb3Config.web3ProviderEndpoints[0]!
+      const discoveryweb3endpoint = `${this.discoveryProvider?.discoveryProviderEndpoint}/chain`
+      const web3endpoint = this.discoveryProvider !== null ? discoveryweb3endpoint : gatewayweb3endpoint
+      this.web3 = new Web3(this.provider(web3endpoint, 10000))
```

Comment:
So if I'm understanding we reset web3 each time in this case to make sure it's using a valid discovery node?

Reply:
yeah but the only usage of this call is during the init step, mainly because the discovery provider is initialized after the web3manager so we have to set it later as opposed to in the constructor